docs/designs/schema-validation.md

```diff
@@ -0,0 +1,10 @@
+# Schema Validation
+Schema validation will be executed during json/yaml deserialization, we use C# class to define the schema. Below are the attributes we use with the equivalent keywords in json schema.
+
+| Attribute | Equivalent Json Schema Keyword | Description | Namespace |
+| - | - | - | - |
+| MinLengthAttribute | minItems | An array or string instance is valid against if its size is greater than, or equal to, the value of this keyword. | System.ComponentModel.DataAnnotations |
+| MaxLengthAttribute | maxItems | An array or string instance is valid against if its size is less than, or equal to, the value of this keyword. | System.ComponentModel.DataAnnotations |
+| RegularExpressionAttribute | pattern | A string instance is considered valid if the regular expression matches the instance successfully. | System.ComponentModel.DataAnnotations |
+| JsonRequiredAttribute(Not RequiredAttribute) | required | An object instance is valid against this keyword if every required property exists in the instance. | Newtonsoft.Json |
```

Comment:
>(Not RequiredAttribute) [](start = 23, length = 23)

What about move this to description, and add some information about why not use this?

Reply:
Sounds good, trying to elaborate more about why

---
In reply to: [207756794](https://github.com/dotnet/docfx/pull/3151#discussion_r207756794) [](ancestors = 207756794)